test/src/test/groovy/org/openremote/test/energy/EnergyOptimisationAssetTest.groovy

```diff
@@ -128,8 +127,8 @@ class EnergyOptimisationAssetTest extends Specification implements ManagerContai
                     managerTestSetup.electricityBatteryAssetId,
                     ElectricityAsset.POWER_SETPOINT.name,
                     new AssetDatapointIntervalQuery(
-                            LocalDateTime.ofInstant(optimisationTime, ZoneId.systemDefault()).plus((long)(optimiser.intervalSize * 60), ChronoUnit.MINUTES),
-                            LocalDateTime.ofInstant(optimisationTime, ZoneId.systemDefault()).plus(1, ChronoUnit.DAYS).minus((long)(optimiser.intervalSize * 60), ChronoUnit.MINUTES),
+                            optimisationTime.atZone(ZoneId.systemDefault()).plus((long)(optimiser.intervalSize * 60), ChronoUnit.MINUTES).toLocalDateTime(),
+                            optimisationTime.atZone(ZoneId.systemDefault()).plus(1, ChronoUnit.DAYS).minus((long)(optimiser.intervalSize * 60), ChronoUnit.MINUTES).toLocalDateTime(),
```

Comment:
It might be an idea to add an `Instant` constructor to `AssetDatapointIntervalQuery` for ease of use so there will be less code to write/read.